_overviews/tutorials/binary-compatibility-for-library-authors.md

```diff
@@ -0,0 +1,140 @@
+in JAR files. The bytecode is what we refer to as the **Binary** format.
+
+When application or library code is compiled, their bytecode invokes named references of classes/methods from their dependencies instead of including the dependencies' actual bytecode
+(unless inlining is explicitly requested). During runtime, the JVM classloader will search through the provided class files for classes/methods referenced by and invoke them.
+
+Let's illustrate with an example:  
+
+We got an application `App` that depends on `A` which itself depends on library `C`. When starting the application we need to provide the class files
+for all of `App`, `A` and `C` (something like `java -cp App.jar:A.jar:C.jar:. MainClass`). If we did not provide `C.jar`, or if we provided a `C.jar` that does not contain certain classes/methods
+which `A` expected to exist in library `C`, we will get an exception when our code attempt to invoke the missing classes/methods.
+
+This is what we call **Binary Incompatibility Errors** - The bytecode interface used for compilation differs and is incompatible with the bytecode provided during runtime.
+
+## What are Evictions, Source Compatibility and Binary Compatibility?
+
+Since the classloader only loads the first match of a class, having multiple versions of the same library in the classpath is redundant. 
+Therefore when deciding which JARs to use for compilation, SBT only selects one version from each library (by default the highest), 
+and all other versions of the same library are **evicted**. When packaging applications, the same versions of libraries that was used for compiling the 
+application is packaged and used during runtime.
+
+Two library versions are said to be **Source Compatible** if switching one for the other does not incur any compile errors. For example, If we can switch from `v1.0.0` of a dependency to `v2.0.0` and
+recompile our code without causing any compilation errors, `v2.0.0` is said to be source compatible with `v1.0.0`.
+
+Two library versions are said to be **Binary Compatible** if the compiled bytecode of these versions are compatible. Using the example above, removing a class will render two version
+binary incompatible too, as the compiled bytecode for v2.0.0 will no longer contain the removed class.
+
+When talking about two versions being compatible, the direction matters too. If we can use `v2.0.0` in place of `v1.0.0`, `v2.0.0` is said to be **backwards compatible** with `v1.0.0`. Conversely,
+if we say that any library release of `v1.x.x` will be forwards compatible, we can use `v1.0.0` anywhere where `v1.1.0` was originally used.
+For the rest of the guide, when the 'compatible' is used we mean backwards compatible, as it is the more common case of compatibility guarantee.
+
+An important note to make is that while breaking source compatibility normally results in breaking binary compatibility, they are actually orthogonal 
```

Comment:
I think you can reword this with one of my previous suggestions and make this paragraph much nicer to read? 👍